internal/config/server/config.go

```diff
@@ -86,14 +86,17 @@ func setViperStructDefaults(v *viper.Viper, prefix string, s any) {
 		}
 		valueName := strings.ToLower(prefix + field.Tag.Get("mapstructure"))
 
+		// Extract a default value the `default` struct tag
+		// we don't support all value types yet, but we can add them as needed
+		value := field.Tag.Get("default")
+
 		if field.Type.Kind() == reflect.Struct {
-			setViperStructDefaults(v, valueName+".", reflect.Zero(field.Type).Interface())
+			if value != "{}" {
+				setViperStructDefaults(v, valueName+".", reflect.Zero(field.Type).Interface())
+			}
```

Comment:
> This means that setting "{}" as the default for a struct clears any underlying defaults, so the fields end up being must-set?

Yes, they don't anymore have the values indicated by the `default` struct tag. Though when unmarshalled from viper, non-set values are defaulted to zero values. 

> Can you leave a TODO to unpack the value from JSON and ignore the underlying struct annotations?

Do you mean using the `json` annotation instead of `default`? Or do you mean these values should be present in the config file?

Reply:
I meant that it would be nice to be able to say:

```golang
type Config struct {
	Events EventConfig `mapstructure:"events" default:"{\"driver\":\"sql\"}"`
}
```

Though, having typed it out, the use of `"` for both JSON and structure tags is annoying.